parlai/core/torch_agent.py

```diff
@@ -227,7 +237,7 @@ def maintain_dialog_history(self, observation, reply='',
         """
 
         def parse(txt, splitSentences):
-            if dict is not None:
```

Comment:
actually I think we can remove this check, currently we expect agents to call super().__init__ and in the init we set up self.dict